src/main/java/org/gridsuite/sensitivityanalysis/server/service/nonevacuatedenergy/NonEvacuatedEnergyInputBuilderService.java

```diff
@@ -0,0 +1,514 @@
+            .forEach(generatorLimitByType -> {
+                // build sensitivity variable sets from generators limits in input data
+                List<SensitivityVariableSet> vInjectionsSets = buildSensitivityVariableSets(
+                    context.getNetworkUuid(), context.getVariantId(), network, reporter,
+                    List.of(IdentifiableType.GENERATOR),
+                    generatorLimitByType.getGenerators(),
+                    SensitivityAnalysisInputData.DistributionType.PROPORTIONAL_MAXP);
+                context.getNonEvacuatedEnergyInputs().addSensitivityVariableSets(vInjectionsSets);
+
+                // we store the cappings generators by energy source
+                context.getNonEvacuatedEnergyInputs().getCappingsGenerators().put(generatorLimitByType.getEnergySource(),
+                    vInjectionsSets.stream().flatMap(v -> v.getVariablesById().keySet().stream()).toList());
+
+                // build sensitivity factors from the variable sets (=set of generators), the contingencies and the monitored branches sets
+                monitoredBranches.stream()
+                    .filter(NonEvacuatedEnergyMonitoredBranches::isActivated)  // we keep only activated monitored branches set
+                    .forEach(branches -> {
+                        List<SensitivityFactor> fInjectionsSet = buildSensitivityFactorsFromVariablesSets(
+                            context, network, reporter,
+                            List.of(IdentifiableType.LINE, IdentifiableType.TWO_WINDINGS_TRANSFORMER),
+                            branches,
+                            vInjectionsSets,
+                            context.getNonEvacuatedEnergyInputs().getContingencies(),
+                            SensitivityVariableType.INJECTION_ACTIVE_POWER);
+                        context.getNonEvacuatedEnergyInputs().addSensitivityFactors(fInjectionsSet);
+                    });
+            });
+
+        // build inputs for the sensitivities in A for each generator (similar to sensitivity analysis computation with injections)
+        generatorsLimitByType.stream()
+            .filter(NonEvacuatedEnergyGeneratorLimitByType::isActivated)  // we keep only activated generators limits
+            .forEach(generatorLimitByType -> {
+                // build sensitivity factors from the variables (=generators), the contingencies and the monitored branches sets
+                monitoredBranches.stream()
+                    .filter(NonEvacuatedEnergyMonitoredBranches::isActivated) // we keep only activated monitored branches set
+                    .forEach(branches -> {
+                        // build the sensitivity factors for each monitored branch
+                        List<SensitivityFactor> fInjectionsSet = buildSensitivityFactors(
+                            context, network, reporter,
+                            List.of(IdentifiableType.LINE, IdentifiableType.TWO_WINDINGS_TRANSFORMER),
```

Comment:
Can we put this restriction to Line and 2WT in a separate explicit entry that we can maintain easily?